managedtenants/schemas/metadata.schema.yaml

```
@@ -82,6 +82,41 @@ properties:
   pullSecret:
     type: string
     format: printable
+    description: "'pullSecret' is deprecated for now. Please use the 'secrets' and 'pullSecretName' fields instead."
+  secrets:
+    type: array
+    description: "List of secrets that are required by the addon operator."
```

```suggestion
    description: "List of secrets that are required by the addon."
```

Because the "addon" depends on these secrets for its expected functioning. ADO just applies this secrets, doesn't depend on them necessarily:)